contracts/cities/nyc/local/newyorkcitycoin-token-v2.clar

```diff
@@ -0,0 +1,222 @@
+(define-data-var tokenActivated bool false)
+
+;; core contract states
+(define-constant STATE_DEPLOYED u0)
+(define-constant STATE_ACTIVE u1)
+(define-constant STATE_INACTIVE u2)
+
+;; one-time function to activate the token
+(define-public (activate-token (coreContract principal) (stacksHeight uint))
+  (let
+    (
+      (coreContractMap (try! (contract-call? .newyorkcitycoin-auth get-core-contract-info coreContract)))
+    )
+    (asserts! (is-eq (get state coreContractMap) STATE_ACTIVE) ERR_UNAUTHORIZED)
+    (asserts! (not (var-get tokenActivated)) ERR_TOKEN_ALREADY_ACTIVATED)
+    (var-set tokenActivated true)
+    (var-set coinbaseThreshold1 (+ stacksHeight TOKEN_BONUS_PERIOD TOKEN_EPOCH_LENGTH))        ;; 35,000 blocks
+    (var-set coinbaseThreshold2 (+ stacksHeight TOKEN_BONUS_PERIOD (* u2 TOKEN_EPOCH_LENGTH))) ;; 85,000 blocks
+    (var-set coinbaseThreshold3 (+ stacksHeight TOKEN_BONUS_PERIOD (* u3 TOKEN_EPOCH_LENGTH))) ;; 185,000 blocks
+    (var-set coinbaseThreshold4 (+ stacksHeight TOKEN_BONUS_PERIOD (* u4 TOKEN_EPOCH_LENGTH))) ;; 385,000 blocks
+    (var-set coinbaseThreshold5 (+ stacksHeight TOKEN_BONUS_PERIOD (* u5 TOKEN_EPOCH_LENGTH))) ;; 785,000 blocks
+    (ok true)
+  )
+)
+
+;; return coinbase thresholds if token activated
+(define-read-only (get-coinbase-thresholds)
+  (let
+    (
+      (activated (var-get tokenActivated))
+    )
+    (asserts! activated ERR_TOKEN_NOT_ACTIVATED)
+    (ok {
+      coinbaseThreshold1: (var-get coinbaseThreshold1),
+      coinbaseThreshold2: (var-get coinbaseThreshold2),
+      coinbaseThreshold3: (var-get coinbaseThreshold3),
+      coinbaseThreshold4: (var-get coinbaseThreshold4),
+      coinbaseThreshold5: (var-get coinbaseThreshold5)
+    })
+  )
+)
+
+;; CONVERSION
+
+(define-public (convert-to-v2)
+  (let
+    (
+      (balanceV1 (unwrap! (contract-call? .newyorkcitycoin-token get-balance tx-sender) ERR_V1_BALANCE_NOT_FOUND))
+    )
+    ;; verify positive balance
+    (asserts! (> balanceV1 u0) ERR_V1_BALANCE_NOT_FOUND)
+    ;; burn old
+    (try! (contract-call? .newyorkcitycoin-token burn balanceV1 tx-sender))
+    (print {
+      burnedV1: balanceV1,
+      mintedV2: (* balanceV1 MICRO_CITYCOINS),
+      tx-sender: tx-sender,
+      contract-caller: contract-caller
+    })
+    ;; create new
+    (ft-mint? newyorkcitycoin (* balanceV1 MICRO_CITYCOINS) tx-sender)
+  )
+)
+
+;; UTILITIES
+
+(define-data-var tokenUri (optional (string-utf8 256)) (some u"https://cdn.citycoins.co/metadata/newyorkcitycoin.json"))
+
+;; set token URI to new value, only accessible by Auth
+(define-public (set-token-uri (newUri (optional (string-utf8 256))))
+  (begin
+    (asserts! (is-authorized-auth) ERR_UNAUTHORIZED)
+    (ok (var-set tokenUri newUri))
+  )
+)
+
+;; mint new tokens, only accessible by a Core contract
+(define-public (mint (amount uint) (recipient principal))
+  (let
+    (
+      (coreContract (try! (contract-call? .newyorkcitycoin-auth get-core-contract-info contract-caller)))
```

Comment:
Why v2 is not used here?

Reply:
fixed in https://github.com/citycoins/contracts/pull/190/commits/f18d325c5d6effec0e70a2fba386f401eeb8e74e